File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int max_water = 0;
        int left = 0, right = heights.size() - 1;
        while (left < right) {
            int left_h = heights[left], right_h = heights[right];
            int area = (right - left) * min(right_h, left_h);
            max_water = max(area, max_water);
            if (left_h > right_h) {
                right--;
            }
            else {
                left++;
            }
        }
        return max_water;
    }
};

File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        longest = ""
        for i in range(len(s)):
            left, right = i, i
            while left > -1 and right < len(s):
                if s[left] == s[right]:
                    if right - left + 1 > len(longest):
                        longest = s[left: right+1]
                    left -= 1
                    right += 1
                else:
                    break

            left, right = i, i+1
            while left > -1 and right < len(s):
                if s[left] == s[right]:
                    if right - left + 1 > len(longest):
                        longest = s[left: right+1]
                    left -= 1
                    right += 1
                else:
                    break
        return longest